docs/source/uploading-files.rst

```diff
@@ -0,0 +1,82 @@
+Uploading Files
+===============
+To upload files for analysis, you need only upload them to the BinaryAlert S3 bucket. The S3 bucket name is of the form
+
+.. code-block:: none
+
+  YOUR.NAME.PREFIX.binaryalert-binaries.REGION
+
+When uploading to S3, any object metadata you set will be included in all match alerts. In addition, if there is a ``filepath`` metadata key, BinaryAlert will make the filepath :ref:`external-variables` available to the YARA rules.
+
+Uploaded files are persisted indefinitely so that BinaryAlert can retroactively analyze all files with every rule update. The S3 bucket has both `access logs <http://docs.aws.amazon.com/AmazonS3/latest/dev/ServerLogs.html>`_ and `object versioning <http://docs.aws.amazon.com/AmazonS3/latest/dev/ObjectVersioning.html>`_ enabled.
+
+
+CarbonBlack Downloader
+----------------------
+If you use CarbonBlack Enterprise Response, you can enable BinaryAlert's optional downloader Lambda function. The downloader takes an MD5 as input and copies the file from CarbonBlack into BinaryAlert's S3 bucket (with the appropriate metadata). To enable it:
+
+.. code-block:: none
+
+  $ ./manage.py configure
+  AWS Region (us-east-1):
+  Unique name prefix, e.g. "company_team": your_unique_prefix
+  Enable the CarbonBlack downloader? (no): yes
+  CarbonBlack URL: https://your.carbonblack.url
+  CarbonBlack API token (only needs binary read access):
+
+.. warning:: The API token only needs access to read binaries; do not use a token with admin privileges, do not allow other users to share the same token, and be sure to regularly rotate the token.
+
+.. note:: The API token will not be shown on screen and BinaryAlert will create a new KMS key to encrypt the credentials before saving them to the ``terraform.tfvars`` configuration file. The downloader (and no other component) is authorized to decrypt the credentials with the generated key.
+
+Once BinaryAlert is deployed with the new configuration, invoking the downloader is as simple as:
```

Comment:
this is confusing because it makes it seem like the user manually has to run this python code every time they want to re-analyze, whereas its already supported via the cli?

Reply:
This is the downloader, not the analyzer. If users aren't using the CarbonBlack downloader, they will indeed need that Python code. I'll move it to the `real-time invocations` section